### PR TITLE
TST: Test type issue fix in empty groupby from DataFrame with categorical

### LINKED ISSUE #9614: Type issue in empty groupby from DataFrame with categorical
In a `DataFrame` without a categorical, the following comparisons work as expected:

``` python
df = pd.DataFrame({'id': [None] * 3, 'spam': [None] * 3})
df['spam'] == 'spam'
df.groupby('id').first()['spam'] == 'spam'
```

However, when a column is `Categorical`, a `groupby` on the all-null column behaves unexpected:

``` python
df['spam'] = df['spam'].astype('category')
df['spam'] == 'spam'  # works as expected
df.groupby('id').first()['spam'] == 'spam'  # raises TypeError: invalid type comparison
```

Looks like the groupby converts all types in the group to `float64`:

``` python
>>> df.groupby('id').first().info()
<class 'pandas.core.frame.DataFrame'>
Int64Index: 0 entries
Data columns (total 1 columns):
spam    0 non-null float64
dtypes: float64(1)
memory usage: 0.0 bytes
```

### PR DESCRIPTION
**Pull request by louishuynh**
TST: Test type issue fix in empty groupby from DataFrame with categorical

closes #9614

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -1825,6 +1825,18 @@ def test_groupby_empty_list_raises():
         df.groupby([[]])
 
 
+def test_groupby_with_empty_category():
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: can you move this to groupby/test_categorical.py

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -1825,6 +1825,18 @@ def test_groupby_empty_list_raises():
...
+    # GH-9614
+    df = pd.DataFrame({"id": [None] * 3, "spam": [None] * 3})
+    df["spam"] = df["spam"].astype("category")
+    df["spam"] == "spam"  # works as expected
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: can you assert this result here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_groupby.py (hunk 1 above)
Comment (louishuynh) [reply pandas/tests/groupby/test_groupby.py]: Done, thanks!

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (louishuynh) [reply pandas/tests/groupby/test_groupby.py]: I've refactored to what I think is more clear & explicit. 
Please approve fine or let me know more asserts still needed, thanks!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @louishuynh

**Pull request merged**